portality/models/article.py

```diff
@@ -271,6 +292,9 @@ def _generate_index(self):
         issns += cbib.get_identifiers(cbib.P_ISSN)
         issns += cbib.get_identifiers(cbib.E_ISSN)
 
+        # get the issn from the journal bibjson
+        issns.append(cbib.journal_issn)
```

Comment:
how does this relate to the two lines above it? .get_identifiers() for PISSN and EISSN . Sounds like this would produce a list of duplicate ISSNs, e.g. if PISSN is 0123-4567 and there is no EISSN, then executing lines 292-296 (as per your new code) will result in an issns list of ['0123-4567', '0123-4567'] as the PISSN will be returned by both .get_identifiers(cbib.P_ISSN) and cbib.journal_issn. Perhaps another deduplication? Or removing lines 292 and 293 as they're potentially superseded by 296?